src/HDWallet.h

```diff
@@ -81,6 +81,9 @@ class HDWallet {
     /// Returns the private key at the given derivation path.
     PrivateKey getKey(const TWCoinType coin, const DerivationPath& derivationPath) const;
 
+    /// Returns the private stark key at the given derivation path from a valid eth signature.
+    PrivateKey getStarkeyFromSignature(const DerivationPath& derivationPath, const std::string& signature) const;
+
```

Comment:
I think we absolutely need the `getStarkeyFromSignature`  in HD class, it's the equivalent of `getKey(TWCurveStarkex)` to get a `PrivateKey` - regarding EIP2645 it can be used standalone, but make it easier to use from an HDWallet class. From an end-user perspective, it's as follow:

```swift
// In real app you retrieve your HDWallet from the KeyStore
let wallet = HDWallet(mnemonic: "obscure opera favorite shuffle mail tip age debate dirt pact cement loyal", passphrase: "")!
let derivationPath = wallet.getEip2645Path(ethAddress: "0xd0972E2312518Ca15A2304D56ff9cc0b7ea0Ea37", layer: "starkex", application: "immutablex", index: "1")
let starkDerivationPath = DerivationPath(string: starkDerivationPath)!
let ethPrivateKey = wallet.getKeyForCoin(coin: CoinType.ethereum)
let ethMsg = "Only sign this request if you’ve initiated an action with Immutable X."
let ethSignature = EthereumMessageSigner.signMessage(privateKey: ethPrivateKey, message: ethMsg)
let starkPrivateKey = wallet.getStarkKeyFromSignature(derivationPath: derivationPath, signature: ethSignature)
```

It make sense that you retrieve privateKey from an HDWallet, it would be inconsistent to get the private key from somewhere else to be honest.

> I'm not sure if logically they belong to HDWallet, which should be kept to HD wallet functionality only

Actually, HDWallet is the class that make the usage of DerivationPath to get a PrivateKey, that doesn't change with this new method, it's just unusual to get a PrivateKey from a signature, but in this case, it's how it works layer2 - as you can see we also offer the possibility to get the private key from the mnemonic and the curve, but it will be incompatible if an user already register on https://market.immutable.com/ as it uses the signature method.

Reply:
It's should be possible to have from ethereum interface:

`EIP2645.getPath()`
`StarkWare.getStarkKeyFromSignature()`

since `bip32DeriveRawSeed` is a static function.

Reply:
a good idea to have `StarkWare` interface and move all related methods there

Reply:
Implemented